### DISCUSSION TITLE
Can I recommend a code running component base on sandpack?

### DISCUSSION THREAD

**Discussion by jerrywu001** [vuejs/vitepress#2376] [General]
Recently I wrote a Vitepress plugin that directly runs scripts written in MD syntax, and supports major languages, and Vite.

- base on [codesandbox](https://codesandbox.io/)
- [supports major languages](https://vitepress-sandbox.js-bridge.com/get-started/introduction.html): vue, react, angular, svelte, vanilla, solid, lit, astro, and tests
- [support multi themes](https://vitepress-sandbox.js-bridge.com/basic-usage/themes.html)
- [support vite4](https://vitepress-sandbox.js-bridge.com/vite-templates/vite-vue-ts.html)
- [run scripts written by md directly](https://vitepress-sandbox.js-bridge.com/basic-usage/rtl.html)
- support [use snippets](https://vitepress-sandbox.js-bridge.com/basic-usage/snippets.html)
- [support dark mode](https://vitepress-sandbox.js-bridge.com/basic-usage/themes.html#custom-theme)
- nice style default

**docs**

https://www.npmjs.com/package/vitepress-plugin-sandpack

**npm package**

https://www.npmjs.com/package/vitepress-plugin-sandpack

**Some screenshots**

![image](https://github.com/vuejs/vitepress/assets/12525415/f94de3f6-b562-4f6f-a963-3e9b0cad16dd)

![image](https://github.com/vuejs/vitepress/assets/12525415/6a4bad4b-6c78-401e-b1d3-f10d5c78f9f5)

![image](https://github.com/vuejs/vitepress/assets/12525415/8774725e-2045-489f-921d-7227cdb5d2ce)

![image](https://github.com/vuejs/vitepress/assets/12525415/b6f5923e-9004-4211-8cc0-257cf79727a2)

![image](https://github.com/vuejs/vitepress/assets/12525415/6b47719a-60de-4ef2-8e03-1632071133bf)

**Comment by HarrySan001** [comment]
awesome!
